poleno/attachments/utils.py

```diff
@@ -5,6 +5,18 @@
 from poleno import datacheck
 
 
+def attachment_file_is_working(attachment):
+    if not attachment.file:
+        return None
+    try:
+        try:
+            attachment.file.open(u'rb')
+        finally:
+            attachment.file.close()
+    except IOError:
+        return False
+    return True
```

Comment:
Trochu ma vyrusuje, ze na kontrolu, ci subor existuje pouzivame `attachment.file.open(u'rb')`. Ale na samotne precitanie suboru pouzivame `open(path, u'rb')`, kde `path` je `os.path.join(settings.MEDIA_ROOT, attachment.file.name)`.

Zrejme je to ekvivalentne. Ale nie je good practice na overenie pouzit iny postup, nez potom pri realnom pouziti.

Nebolo by lepsie skor priamo v `download` view nieco ako:

```
if not attachment.file:
  raise Http404()
path = os.path.join(settings.MEDIA_ROOT, attachment.file.name)
try:
  return send_file_response(...)
except (IOError, OSError):
  raise Http404()
```

Btw, pls zisti, ktore z `IOError`/`OSError` pripadne oba alebo aj nejake ine potrebujeme? Ta exception co nam nastala bola:
```
OSError: Not a regular file: /var/www/chcemvediet/chcemvediet/media/
```

Este rozmyslam, ze v pripadne ze v `send_file_response(...)`  nastane ta exception a my ju odchytime, zislo by sa ju zalogovat. Kedze tato chyba by realne nastavat nemala. Ak `file.name` nie je prazdny a subor neexistuje, alebo sa neda precitat, tak je nieco zle. 

Ako sa loguju errory pozri napriklad `logger.error(...)` v `poleno/attachments/models.py`.

Reply:
`OSError: Not a regular file: /var/www/chcemvediet/chcemvediet/media/` nam uz nenastane, to je osetrene podmienkou `if not attachment.file: raise Http404()`.

Reply:
Chybu `OSError: Not a regular file: /var/www/chcemvediet/chcemvediet/media/` zachyti if podmienka `    if not attachment.file: return None`

Reply:
Vznikali nam dve chyby: 
1) OSError Not a regular file: `/chcemvediet/media/attachments_finalizations/`
 ta vznikla ak bolo `attachment.file.name` prazdny string. Tuto chybu vyvolalo 
  ```
def send_file_response(...):
      ...
      if not stat.S_ISREG(statobj.st_mode):
        raise OSError(u'Not a regular file: {}'.format(path))
```

2) OSError at [Errno 2] No such file or directory: `chcemvediet/media/attachment_finalizations/29RVCWztPw`
ta vznikla ak v `attachment.file` malo svoju hodnotu, ale subor fyzicky neexistoval, subor je vymazany. Tuto chybu vyvolalo:
```
def send_file_response(...):
      statobj = os.stat(path)
      ...
```

Takze nam to bude fungovat aj bez prvej `if` podmienky.:

```
path = os.path.join(settings.MEDIA_ROOT, attachment.file.name)
    try:
        return send_file_response(request, path, filename or attachment.name, attachment.content_type)
    except OSError:
        raise Http404()
```

`IOError` sa vratil vtedy ked sme citali zo suboru `attachment.file.open('rb')`, ale jedna sa o tu istu chybu. Na to ci attachment existujeme mame vytvorene `datacheck`. Mam  aj napriek to mu logovat chybu v `download` view, ked uz mame datachecks?